### PR TITLE
Add intelligence command to the CLI client

### PR DESCRIPTION
**Pull request by jaegeral**
This pull requests adds a new command int he Timesketch CLI called `intelligence`.

```
timesketch intelligence -h
Usage: timesketch intelligence [OPTIONS] COMMAND [ARGS]...

  Manage intelligence within a sketch.

Options:
  -h, --help  Show this message and exit.

Commands:
  add   Add intelligence to a sketch.
  list  List all intelligence.
```

Add an ioc:
```
timesketch intelligence add -h
Usage: timesketch intelligence add [OPTIONS]

  Add intelligence to a sketch.

  A sketch can have multiple intelligence entries. Each entry consists of an
  indicator, a type and a list of tags.

  Reference: https://timesketch.org/guides/user/intelligence/

  Args:     ctx: Click context object.     ioc: IOC value.     inteltype: Type of
  the intelligence. This is defined in the ontology file.         If a string
  doesn't match any of the aforementioned IOC types,         the type will
  fall back to other.     tags: Comma separated list of tags.

Options:
  --ioc TEXT   IOC value.  [required]
  --inteltype TEXT  Type of the intelligence.
  --tags TEXT  Comma separated list of tags.
  -h, --help   Show this message and exit.
```

Example:
```
timesketch --sketch 2  intelligence add --ioc 8.8.4.6 --inteltype ipv4
Intelligence added: 8.8.4.6
```

List intelligence:
```
timesketch intelligence list -h
Usage: timesketch intelligence list [OPTIONS]

  List all intelligence.

  Args:     ctx: Click context object.     header: Include header in output.
  (default is to show header)     columns: Comma separated list of columns to
  show. (default: ioc,type)             Other options: externalURI, tags

Options:
  --header / --no-header  Include header in output. (default is to show
                          header))
  --columns TEXT          Comma separated list of columns to show. (default:
                          ioc,type)
  -h, --help              Show this message and exit.
```

Example:
```
timesketch --sketch 2 --output-format json intelligence list
[
    {
        "ioc": "8.8.4.6",
        "tags": [],
        "type": "ipv4"
    }
]
```

Or as csv:
```
timesketch --sketch 2 --output-format csv intelligence list --columns "ioc,tags,type"

ioc,tags,type
8.8.4.6,"",ipv4
```

As text:
```
timesketch --sketch 2 --output-format text intelligence list --columns "ioc,tags,type"
ioc	tags	type
8.8.4.6		ipv4
```

It also adds relevant portions to the documentation.

### CODE REVIEW

**Comment by jaegeral** [comment]
This should be read for a first round of reviews. As with the attributes, we will not burry it in the sketch sub command as more or less everything has to do with a sketch. While in theory you could get anything in this PR done with the already existing `timesketch attributes` command group, the nesting etc make it hard to add IOCs and especially the listing of IOCs can be more specific that way (IMHO).

File: cli_client/python/timesketch_cli_client/commands/intelligence.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    """
+
+    if not columns:
+        columns = "ioc,type"
+
+    columns = columns.split(",")
+
+    output = ctx.obj.output_format
+    sketch = ctx.obj.sketch
+    try:
```
Comment (berggren) [inline cli_client/python/timesketch_cli_client/commands/intelligence.py:55]: This `try` is pretty broad. Can you move this into a guard instead?

**Review by berggren — CHANGES_REQUESTED** [review]

File: cli_client/python/timesketch_cli_client/commands/intelligence.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    A sketch can have multiple intelligence entries. Each entry consists of
+    an indicator, a type and a list of tags.
+
+    Reference: https://timesketch.org/guides/user/intelligence/
+
+    Args:
+        ctx: Click context object.
+        ioc: IOC value.
+        inteltype: Type of the intelligence. This is defined in the ontology file.
```
[the remark this answers is not in the document]
Comment (jaegeral) [reply cli_client/python/timesketch_cli_client/commands/intelligence.py:122]: sure thing

File: cli_client/python/timesketch_cli_client/commands/intelligence.py (hunk 1 above)
Comment (jaegeral) [reply cli_client/python/timesketch_cli_client/commands/intelligence.py:55]: actually I just moved it around the get_intel, that is the only one it matters here

File: cli_client/python/timesketch_cli_client/commands/intelligence.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                    row.append(entry.get(column, ""))
+            click.echo("\t".join(row))
+    elif output == "csv":
+        if header:
+            click.echo(",".join(columns))
+        for entry in intelligence:
+            row = []
+            for column in columns:
+                if column == "tags":
+                    # tags can be multiple values but they should only be
```
Comment (berggren) [inline cli_client/python/timesketch_cli_client/commands/intelligence.py:84]:
```suggestion
                    # Tags can be multiple values but they should only be
```

File: cli_client/python/timesketch_cli_client/commands/intelligence.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        ctx: Click context object.
+        ioc: IOC value.
+        type: Type of the intelligence. This is defined in the ontology file.
+            If a string doesn't match any of the aforementioned IOC types,
+            the type will fall back to other.
+        tags: Comma separated list of tags.
+    """
+    sketch = ctx.obj.sketch
+
+    # create a tags dict from the comma separated list
```
Comment (berggren) [inline cli_client/python/timesketch_cli_client/commands/intelligence.py:130]:
```suggestion
    # Create a tags dict from the comma separated list
```

File: cli_client/python/timesketch_cli_client/commands/intelligence.py [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    if tags:
+        tags = tags.split(",")
+        tags = {tag: [] for tag in tags}
+    else:
+        tags = []
+
+    ioc_dict = {"ioc": ioc, "type": type, "tags": tags}
+    # put the ioc in a nested object to match the format of the API
```
Comment (berggren) [inline cli_client/python/timesketch_cli_client/commands/intelligence.py:138]:
```suggestion
    # Put the ioc in a nested object to match the format of the API
```

**Review by berggren — APPROVED** [review]

**Pull request merged**